app/src/main/java/io/gnosis/safe/ui/settings/owner/details/OwnerDetailsViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package io.gnosis.safe.ui.settings.owner.details
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import io.gnosis.data.repositories.CredentialsRepository
+import io.gnosis.safe.Tracker
+import io.gnosis.safe.notifications.NotificationRepository
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.QrCodeGenerator
+import timber.log.Timber
+import javax.inject.Inject
+
+class OwnerDetailsViewModel
+@Inject constructor(
+    private val credentialsRepository: CredentialsRepository,
+    private val notificationRepository: NotificationRepository,
+    private val tracker: Tracker,
+    private val qrCodeGenerator: QrCodeGenerator,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<OwnerDetailsState>(appDispatchers) {
+
+    override fun initialState() = OwnerDetailsState(ViewAction.None)
+
+    fun loadOwnerDetails(address: Solidity.Address) {
+        safeLaunch {
+
+            val ownerName = credentialsRepository.owner(address)!!.name
+
+            val qrCode = runCatching {
+                qrCodeGenerator.generateQrCode(
+                    address.asEthereumAddressChecksumString(),
+                    512,
```

Comment:
Should this be an extracted constant? Does it have a relationship with the `owner_qr_code` views width and height?

Reply:
it's size of a bitmap. I would address this in scope of export owner screen, as there we have yet another qr code :)